flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/HiveCatalog.java

```diff
@@ -1074,15 +1074,14 @@ public void alterTableStatistics(ObjectPath tablePath, CatalogTableStatistics ta
 	}
 
 	@Override
-	public void alterTableColumnStatistics(ObjectPath tablePath, CatalogColumnStatistics columnStatistics, boolean ignoreIfNotExists) throws TableNotExistException, CatalogException {
+	public void alterTableColumnStatistics(ObjectPath tablePath, CatalogColumnStatistics columnStatistics, boolean ignoreIfNotExists) throws TableNotExistException, CatalogException, TableNotPartitionedException {
 		try {
 			Table hiveTable = getHiveTable(tablePath);
 			// Set table column stats. This only works for non-partitioned tables.
 			if (!isTablePartitioned(hiveTable)) {
-				client.updateTableColumnStatistics(HiveCatalogUtil.createTableColumnStats(hiveTable, columnStatistics.getColumnStatisticsData()));
+				client.updateTableColumnStatistics(HiveStatsUtil.createTableColumnStats(hiveTable, columnStatistics.getColumnStatisticsData()));
 			} else {
-				throw new CatalogException(String.format("Failed to alter partition table column stats of table %s",
-														tablePath.getFullName()));
+				throw new TableNotPartitionedException(getName(), tablePath);
```

Comment:
shouldn't this be a `TablePartitionedException`? not `TableNotPartitionedException`

we need to add a `TablePartitionedException.class`